utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -269,14 +277,18 @@ private fun ArchiveInputStream.unpack(
  * Unpack the [ZipFile]. In contrast to [InputStream.unpackZip] this properly parses the ZIP's central directory, see
  * https://commons.apache.org/proper/commons-compress/zip.html#ZipArchiveInputStream_vs_ZipFile.
  */
-private fun ZipFile.unpack(targetDirectory: File) =
+private fun ZipFile.unpack(targetDirectory: File, filter: (ArchiveEntry) -> Boolean) =
     use { zipFile ->
         val entries = zipFile.entries
+        val skip = skipFilter(
+            { entry -> entry.isDirectory || (entry as ZipArchiveEntry).isUnixSymlink || File(entry.name).isAbsolute },
```

Comment:
I think I'd prefer to not introduce the `skipFilter()` function, but inline the `filter()` call to where the original filtering already happens. At this example, this as the advantage of avoiding the `as ZipArchiveEntry` cast.

Reply:
Done. However, now detekt complains about complex conditions; so I had to add `@Suppress` annotations.